### PR TITLE
documentation: add description for parameters in TransformInput

### PR DESCRIPTION
**Pull request by mwfongAWS**
*Issue #, if available:* 

*Description of changes:*
Add description for parameters in TransformInput()

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by trajanikant — COMMENTED** [review]
/bot run all

**Review by knikure — COMMENTED** [review]
/bot run all

**Comment by mwfongAWS** [comment]
Sorry I had no idea what I was doing before and I got distracted with other projects. I'm hoping this one works out! I made the changes on my laptop and ran the tests and they succeeded after I ran 
tox -e black-check,flake8,pylint,docstyle,sphinx,doc8 —parallel all
then 
tox -e black-format

**Review by knikure — COMMENTED** [review]
/bot run all

File: src/sagemaker/inputs.py [hunk 1]
```diff
@@ -165,6 +165,58 @@ class TransformInput(object):
     """Create a class containing all the parameters.
 
     It can be used when calling ``sagemaker.transformer.Transformer.transform()``
+
+    Args:
+        data (str): The S3 location of the input data that the model can consume.
+        data_type (str): elastic inference accelerator type.
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:171]: Question, 

the arg name is "data_type" but the description is "the type of an elastic inference accelerator", which is not a data type?

File: src/sagemaker/inputs.py [hunk 2]
```diff
@@ -165,6 +165,58 @@ class TransformInput(object):
...
+            (default: None)
+        model_client_config (dict): Configures the timeout and maximum number of retries for
+            processing a transform job invocation.
+
+                * ``'InvocationsTimeoutInSeconds'`` (int) - The timeout value in seconds for an
+                  invocation request. The default value is 600.
+                * ``'InvocationsMaxRetries'`` (int) - The maximum number of retries when invocation
+                  requests are failing.
+
+            (default: ``{'InvocationsTimeoutInSeconds'=600,'InvocationsMaxRetries'=3}``)
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:216]: Is this a correct form? I think dict should be in { 'key': 'value' } pair format.

File: src/sagemaker/inputs.py [hunk 3]
```diff
@@ -165,6 +165,58 @@ class TransformInput(object):
...
+        batch_data_capture_config (dict): Specifies configuration related to batch transform job
+            data capture for use with Amazon SageMaker Model Monitoring.
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:218]: What is "batch transform job data capture"? It seems that more explanation is needed breaking this term down.

**Review by mchoi8739 — CHANGES_REQUESTED** [review]

File: src/sagemaker/inputs.py (hunk 1 above)
Comment (mchoi8739) [reply src/sagemaker/inputs.py:171]: Actually should it be "The data type for batch transform job"?

File: src/sagemaker/inputs.py [hunk 4]
```diff
@@ -165,6 +165,63 @@ class TransformInput(object):
     """Create a class containing all the parameters.
 
     It can be used when calling ``sagemaker.transformer.Transformer.transform()``
+
+    Args:
+        data (str): The S3 location of the input data that the model can consume.
+        data_type (str): The data type for an elastic inference accelerator.
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:171]: Actually should it be "The data type for a batch transform job"? "an elastic inference accelerator" is only a device.

File: src/sagemaker/inputs.py [hunk 5]
```diff
@@ -165,6 +165,63 @@ class TransformInput(object):
     """Create a class containing all the parameters.
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:165]:
```suggestion
    """Creates a class containing parameters for configuring input data for a batch tramsform job.
```

File: src/sagemaker/inputs.py [hunk 6]
```diff
@@ -165,6 +165,63 @@ class TransformInput(object):
...
+        model_client_config (dict): Configures the timeout and maximum number of retries for
+            processing a transform job invocation.
+
+                * ``'InvocationsTimeoutInSeconds'`` (int) - The timeout value in seconds for an
+                  invocation request. The default value is 600.
+                * ``'InvocationsMaxRetries'`` (int) - The maximum number of retries when invocation
+                  requests are failing.
+
+            (default: ``{600,3}``)
+        batch_data_capture_config (dict): Specifies configuration related to batch transform job
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:219]:
```suggestion
        batch_data_capture_config (An object of `BatchDataCaptureConfig <https://sagemaker.readthedocs.io/en/stable/api/utility/inputs.html#sagemaker.inputs.BatchDataCaptureConfig>`_): 
        Specifies configuration to capture data from a batch transform job
```

File: src/sagemaker/inputs.py [hunk 7]
```diff
@@ -165,6 +165,63 @@ class TransformInput(object):
...
+            data capture for use with Amazon SageMaker Model Monitoring. For more information about
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:220]:
```suggestion
            for use with Amazon SageMaker Model Monitoring. For more information,
```

File: src/sagemaker/inputs.py [hunk 8]
```diff
@@ -165,6 +165,63 @@ class TransformInput(object):
...
+            batch transform job data capture, see `Capture data from batch transform job
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:221]:
```suggestion
            see `Capture data from batch transform job
```

**Review by mchoi8739 — CHANGES_REQUESTED** [review]

**Review by mchoi8739 — APPROVED** [review]

**Review by knikure — COMMENTED** [review]
/bot run all

File: src/sagemaker/inputs.py [hunk 9]
```diff
@@ -162,9 +162,68 @@ class CreateModelInput(object):
...
+            processing a transform job invocation.
+
+                * ``'InvocationsTimeoutInSeconds'`` (int) - The timeout value in seconds for an
+                  invocation request. The default value is 600.
+                * ``'InvocationsMaxRetries'`` (int) - The maximum number of retries when invocation
+                  requests are failing.
+
+            (default: ``{600,3}``)
+        batch_data_capture_config (dict, an object of `BatchDataCaptureConfig 
+        <https://sagemaker.readthedocs.io/en/stable/api/utility/inputs.html#sagemaker.inputs.BatchDataCaptureConfig>`_)
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:220]:
```suggestion
            <https://sagemaker.readthedocs.io/en/stable/api/utility/inputs.html#sagemaker.inputs.BatchDataCaptureConfig>`_)
```

File: src/sagemaker/inputs.py [hunk 10]
```diff
@@ -162,9 +162,68 @@ class CreateModelInput(object):
...
+            processing a transform job invocation.
+
+                * ``'InvocationsTimeoutInSeconds'`` (int) - The timeout value in seconds for an
+                  invocation request. The default value is 600.
+                * ``'InvocationsMaxRetries'`` (int) - The maximum number of retries when invocation
+                  requests are failing.
+
+            (default: ``{600,3}``)
+        batch_data_capture_config (dict, an object of `BatchDataCaptureConfig
+            <https://sagemaker.readthedocs.io/en/stable/api/utility/inputs.html#sagemaker.inputs.BatchDataCaptureConfig>`_)
```
Comment (mchoi8739) [inline src/sagemaker/inputs.py:220]:
```suggestion
            <https://sagemaker.readthedocs.io/en/stable/api/utility/inputs.html#sagemaker.inputs.BatchDataCaptureConfig>`_):
```

**Review by knikure — COMMENTED** [review]
/bot run all

**Review by knikure — APPROVED** [review]
/bot run all

**Review by knikure — COMMENTED** [review]
/bot run all

**Pull request merged**